publication-commons/src/test/java/no/unit/nva/publication/service/impl/PublishingServiceTest.java

```diff
@@ -0,0 +1,197 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import no.unit.nva.clients.ChannelClaimDto;
+import no.unit.nva.clients.ChannelClaimDto.ChannelClaim;
+import no.unit.nva.clients.ChannelClaimDto.ChannelClaim.ChannelConstraint;
+import no.unit.nva.clients.ChannelClaimDto.CustomerSummaryDto;
+import no.unit.nva.clients.CustomerDto;
+import no.unit.nva.clients.IdentityServiceClient;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationStatus;
+import no.unit.nva.model.contexttypes.Degree;
+import no.unit.nva.model.contexttypes.Publisher;
+import no.unit.nva.model.instancetypes.degree.DegreeBachelor;
+import no.unit.nva.model.instancetypes.journal.JournalArticle;
+import no.unit.nva.publication.model.business.FilesApprovalThesis;
+import no.unit.nva.publication.model.business.PublishingRequestCase;
+import no.unit.nva.publication.model.business.Resource;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.service.ResourcesLocalTest;
+import no.unit.nva.testutils.RandomDataGenerator;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.ForbiddenException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.core.StringUtils;
+import nva.commons.core.paths.UriWrapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PublishingServiceTest extends ResourcesLocalTest {
+
+    private ResourceService resourceService;
+    private IdentityServiceClient identityServiceClient;
+    private PublishingService publishingService;
+
+    @BeforeEach
+    void setUp() throws NotFoundException {
+        super.init();
+        this.resourceService = getResourceServiceBuilder().build();
+        var ticketService = getTicketService();
+        this.identityServiceClient = mock(IdentityServiceClient.class);
+        when(identityServiceClient.getCustomerById(any())).thenReturn(customerWithWorkflow(
+            REGISTRATOR_PUBLISHES_METADATA_ONLY.getValue()));
+        this.publishingService = new PublishingService(resourceService, ticketService, identityServiceClient);
+    }
+
+    @Test
+    void shouldPublishResourceWhenPublicationIsPublishable() throws ApiGatewayException {
+        var publication = randomPublication(JournalArticle.class).copy()
+                              .withStatus(PublicationStatus.DRAFT)
+                              .withAssociatedArtifacts(List.of())
+                              .build();
+        var userInstance = UserInstance.fromPublication(publication);
+        publication = Resource.fromPublication(publication).persistNew(resourceService, userInstance);
```

Comment:
IS there a reason to reassign publication with a resource?

Reply:
Yes, the first one is not persisted, it is ugly because Resource.persistNew() returns publication.
I will extract whole thing to a method, so it look prettier.

Reply:
It would be enough to just assign a different variable since it isn't a comparable object anyway